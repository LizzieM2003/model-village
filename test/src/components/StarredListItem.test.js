import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { StarredListItem } from './StarredListItem';
import { StarredList } from './StarredList';
import { removeStarred } from '../actions/starred';

configure({ adapter: new Adapter() });

describe('StarredListItem', () => {
  const influencer = {
    influencer_id: 123,
    influencer_full_name: 'Hannah Ross',
    influencer_instagram_username: 'hross64346',
    influencer_instagram_profile_image:
      'https://randomuser.me/api/portraits/women/62.jpg',
    statistics: {
      followers: 102523,
      engagement: '3.14'
    }
  };

  const mockRemoveStarred = jest.fn();

  const starredListItem = shallow(
    <StarredListItem
      key={influencer.influencer_id}
      {...influencer}
      removeStarred={mockRemoveStarred}
    />
  );

  it('renders correctly', () => {
    expect(starredListItem).toMatchSnapshot();
  });

  describe('when delete button is clicked', () => {
    

    it('calls removeStarred function', () => {
      starredListItem
        .find('.StarredListItem__container__button')
        .simulate('click');
      expect(mockRemoveStarred).toHaveBeenCalledWith(influencer.influencer_id);
    });
  });
});
