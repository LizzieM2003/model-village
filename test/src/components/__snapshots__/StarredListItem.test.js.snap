// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarredListItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    influencer_full_name="Hannah Ross"
    influencer_id={123}
    influencer_instagram_profile_image="https://randomuser.me/api/portraits/women/62.jpg"
    influencer_instagram_username="hross64346"
    removeStarred={[Function]}
    statistics={
        Object {
            "engagement": "3.14",
            "followers": 102523,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="StarredListItem"
>
          <img
                    alt="head shot of influencer"
                    src="https://randomuser.me/api/portraits/women/62.jpg"
          />
          <div
                    className="StarredListItem__details"
          >
                    <h3>
                              Hannah Ross
                    </h3>
                    <p
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "color": "#a9a9a9",
                                                  "display": "flex",
                                                  "fontSize": "14px",
                                                  "marginTop": "10px",
                                                }
                              }
                    >
                              <instagram
                                        color="#a9a9a9"
                                        height="14px"
                                        width="14px"
                              />
                              hross64346
                    </p>
          </div>
          <div
                    className="StarredListItem__followers"
          >
                    <h3>
                              102,523
                    </h3>
                    <p>
                              Followers
                    </p>
          </div>
          <div
                    className="StarredListItem__engagement"
          >
                    <h3>
                              3.14
                              %
                    </h3>
                    <p>
                              Engagement
                    </p>
          </div>
</div>,
        <button
          className="StarredListItem__container__button"
          onClick={[Function]}
>
          x
</button>,
      ],
      "className": "StarredListItem__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="head shot of influencer"
              src="https://randomuser.me/api/portraits/women/62.jpg"
/>,
            <div
              className="StarredListItem__details"
>
              <h3>
                            Hannah Ross
              </h3>
              <p
                            style={
                                          Object {
                                                        "alignItems": "center",
                                                        "color": "#a9a9a9",
                                                        "display": "flex",
                                                        "fontSize": "14px",
                                                        "marginTop": "10px",
                                                      }
                            }
              >
                            <instagram
                                          color="#a9a9a9"
                                          height="14px"
                                          width="14px"
                            />
                            hross64346
              </p>
</div>,
            <div
              className="StarredListItem__followers"
>
              <h3>
                            102,523
              </h3>
              <p>
                            Followers
              </p>
</div>,
            <div
              className="StarredListItem__engagement"
>
              <h3>
                            3.14
                            %
              </h3>
              <p>
                            Engagement
              </p>
</div>,
          ],
          "className": "StarredListItem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "head shot of influencer",
              "src": "https://randomuser.me/api/portraits/women/62.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Hannah Ross
</h3>,
                <p
                  style={
                                    Object {
                                                      "alignItems": "center",
                                                      "color": "#a9a9a9",
                                                      "display": "flex",
                                                      "fontSize": "14px",
                                                      "marginTop": "10px",
                                                    }
                  }
>
                  <instagram
                                    color="#a9a9a9"
                                    height="14px"
                                    width="14px"
                  />
                  hross64346
</p>,
              ],
              "className": "StarredListItem__details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hannah Ross",
                },
                "ref": null,
                "rendered": "Hannah Ross",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <instagram
                      color="#a9a9a9"
                      height="14px"
                      width="14px"
/>,
                    "hross64346",
                  ],
                  "style": Object {
                    "alignItems": "center",
                    "color": "#a9a9a9",
                    "display": "flex",
                    "fontSize": "14px",
                    "marginTop": "10px",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "color": "#a9a9a9",
                      "height": "14px",
                      "width": "14px",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "hross64346",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  102,523
</h3>,
                <p>
                  Followers
</p>,
              ],
              "className": "StarredListItem__followers",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "102,523",
                },
                "ref": null,
                "rendered": "102,523",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Followers",
                },
                "ref": null,
                "rendered": "Followers",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  3.14
                  %
</h3>,
                <p>
                  Engagement
</p>,
              ],
              "className": "StarredListItem__engagement",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "3.14",
                    "%",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "3.14",
                  "%",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Engagement",
                },
                "ref": null,
                "rendered": "Engagement",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "x",
          "className": "StarredListItem__container__button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "x",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="StarredListItem"
>
            <img
                        alt="head shot of influencer"
                        src="https://randomuser.me/api/portraits/women/62.jpg"
            />
            <div
                        className="StarredListItem__details"
            >
                        <h3>
                                    Hannah Ross
                        </h3>
                        <p
                                    style={
                                                Object {
                                                            "alignItems": "center",
                                                            "color": "#a9a9a9",
                                                            "display": "flex",
                                                            "fontSize": "14px",
                                                            "marginTop": "10px",
                                                          }
                                    }
                        >
                                    <instagram
                                                color="#a9a9a9"
                                                height="14px"
                                                width="14px"
                                    />
                                    hross64346
                        </p>
            </div>
            <div
                        className="StarredListItem__followers"
            >
                        <h3>
                                    102,523
                        </h3>
                        <p>
                                    Followers
                        </p>
            </div>
            <div
                        className="StarredListItem__engagement"
            >
                        <h3>
                                    3.14
                                    %
                        </h3>
                        <p>
                                    Engagement
                        </p>
            </div>
</div>,
          <button
            className="StarredListItem__container__button"
            onClick={[Function]}
>
            x
</button>,
        ],
        "className": "StarredListItem__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="head shot of influencer"
                src="https://randomuser.me/api/portraits/women/62.jpg"
/>,
              <div
                className="StarredListItem__details"
>
                <h3>
                                Hannah Ross
                </h3>
                <p
                                style={
                                                Object {
                                                                "alignItems": "center",
                                                                "color": "#a9a9a9",
                                                                "display": "flex",
                                                                "fontSize": "14px",
                                                                "marginTop": "10px",
                                                              }
                                }
                >
                                <instagram
                                                color="#a9a9a9"
                                                height="14px"
                                                width="14px"
                                />
                                hross64346
                </p>
</div>,
              <div
                className="StarredListItem__followers"
>
                <h3>
                                102,523
                </h3>
                <p>
                                Followers
                </p>
</div>,
              <div
                className="StarredListItem__engagement"
>
                <h3>
                                3.14
                                %
                </h3>
                <p>
                                Engagement
                </p>
</div>,
            ],
            "className": "StarredListItem",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "head shot of influencer",
                "src": "https://randomuser.me/api/portraits/women/62.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Hannah Ross
</h3>,
                  <p
                    style={
                                        Object {
                                                            "alignItems": "center",
                                                            "color": "#a9a9a9",
                                                            "display": "flex",
                                                            "fontSize": "14px",
                                                            "marginTop": "10px",
                                                          }
                    }
>
                    <instagram
                                        color="#a9a9a9"
                                        height="14px"
                                        width="14px"
                    />
                    hross64346
</p>,
                ],
                "className": "StarredListItem__details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Hannah Ross",
                  },
                  "ref": null,
                  "rendered": "Hannah Ross",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <instagram
                        color="#a9a9a9"
                        height="14px"
                        width="14px"
/>,
                      "hross64346",
                    ],
                    "style": Object {
                      "alignItems": "center",
                      "color": "#a9a9a9",
                      "display": "flex",
                      "fontSize": "14px",
                      "marginTop": "10px",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "color": "#a9a9a9",
                        "height": "14px",
                        "width": "14px",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "hross64346",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    102,523
</h3>,
                  <p>
                    Followers
</p>,
                ],
                "className": "StarredListItem__followers",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "102,523",
                  },
                  "ref": null,
                  "rendered": "102,523",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Followers",
                  },
                  "ref": null,
                  "rendered": "Followers",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    3.14
                    %
</h3>,
                  <p>
                    Engagement
</p>,
                ],
                "className": "StarredListItem__engagement",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "3.14",
                      "%",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "3.14",
                    "%",
                  ],
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Engagement",
                  },
                  "ref": null,
                  "rendered": "Engagement",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "className": "StarredListItem__container__button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
